packages/base-controller/src/Messenger.ts

```diff
@@ -382,7 +382,7 @@ export class Messenger<
    */
   unsubscribe<EventType extends Event['type']>(
     eventType: EventType,
-    handler: ExtractEventHandler<Event, EventType>,
+    handler: ExtractEventHandler<Event, EventType> | SelectorEventHandler,
```

Comment:
Should we add a test for this to confirm that the types would work? (Same goes for the changes to the `messenger` package.)

Reply:
There was already a test, but it made extensive use of `sinon.stub` with the default `any` as the type parameter, so it wasn't much good for validating the type problem. I've updated the test to minimize the use of `sinon`, ensuring all relevant functions are typed normally.

Updated here: https://github.com/MetaMask/core/pull/6262/commits/000fb9f22d01cd059854a852ed17f6348b26f6bb

This revealed a new type error that I had to resolve as well (by taking @MajorLift 's suggestion above)